packages/astro/src/core/preview/index.ts

```diff
@@ -48,28 +51,22 @@ export default async function preview(config: AstroConfig, { logging }: PreviewO
 		const isRoot = pathname === '/';
 		const hasTrailingSlash = isRoot || pathname.endsWith('/');
 
-		function err(message: string) {
+		function sendError(message: string) {
 			res.statusCode = 404;
 			res.end(notFoundTemplate(pathname, message));
 		};
 
-		switch(true) {
+		switch (true) {
 			case hasTrailingSlash && trailingSlash == 'never' && !isRoot:
-				err('Prohibited trailing slash');
-				break;
-			case !hasTrailingSlash && trailingSlash == 'always' && !isRoot:
```

Comment:
I think this was a bug, where every URL required a trailing slash which would have made URLs like `/favicon.png` fail